vllm/model_executor/model_loader/loader.py

```diff
@@ -39,6 +40,31 @@
 from vllm.platforms import current_platform
 from vllm.utils import is_tpu
 
+
+@contextmanager
+def gpu_loading_context(module: torch.nn.Module):
+    # TODO: change this
+    target_device = torch.device("cuda")
+    original_device_states: Dict[str, torch.device] = {}
+
+    # Store original device states and move parameters to GPU if they're on CPU
+    for name, param in module.named_parameters():
+        if param.device.type == 'cpu':
+            original_device_states[name] = param.device
+            param.data = param.data.to(target_device)
+        # Parameters already on CUDA devices are not touched
+
+    try:
+        yield module
+
+    finally:
+        # Restore parameters to their original devices, ignoring new parameters
+        for name, param in module.named_parameters():
+            if name in original_device_states:
+                param.data = param.data.to(original_device_states[name])
```

Comment:
`to("cpu")` does not support pin memory, which is very important for cpu offloading performance. Instead, can you find out which part of code process weights on GPU, and add `.cuda()` to convert weight from CPU to GPU, before passing it to the quantization kernels? After the quantization, that cuda kernel is gc-released, and be transparent to the model.

Reply:
I think it is intractable to know what parameters need to be on the GPU for each quantization method, so I would rather just respect the device that the original param is on. Is there a reason why I can't just use the same method as you for offloading?